packages/flutter_tools/lib/src/build_system/targets/web.dart

```diff
@@ -286,6 +289,35 @@ class Dart2WasmTarget extends Dart2WebTarget {
     if (compileResult.exitCode != 0) {
       throw Exception(_collectOutput(compileResult));
     }
+    if (compilerConfig.runWasmOpt) {
+      final String wasmOptBinary = artifacts.getArtifactPath(
+        Artifact.wasmOptBinary,
+        platform: TargetPlatform.web_javascript
+      );
+      final File optimizedOutput = environment.buildDir.childFile('main.dart.wasm');
+      final List<String> optimizeArgs = <String>[
+        wasmOptBinary,
+        '-all',
+        '--closed-world',
+        '-tnh',
+        '-O3',
+        '--type-ssa',
+        '--gufa',
+        '-O3',
+        '--type-merging',
+        outputWasmFile.path,
+        '-o',
+        optimizedOutput.path,
+      ];
+      final ProcessResult optimizeResult = await globals.processManager.run(optimizeArgs);
+      if (optimizeResult.exitCode != 0) {
+        throw Exception(_collectOutput(optimizeResult));
+      }
+
+      // Rename the .mjs file not to have the `.unopt` bit
+      final File jsRuntimeFile = environment.buildDir.childFile('main.dart.unopt.mjs');
```

Comment:
where does this unopt mjs file come from?

Reply:
`dart2wasm` produces a js support file with the same name as the output file, except with `.mjs` instead of `.wasm` at the end. Because I'm having the output of the main `dart2wasm` compile process be `main.dart.unopt.wasm` (if I'm planning on running `wasm-opt` after), the support file is named `main.dart.unopt.mjs`, but I really want it to be `main.dart.mjs`, hence why I'm renaming it here. I have a PR out for review that would just let me pass the desired output name for the JS runtime file instead of having to rename it: https://dart-review.googlesource.com/c/sdk/+/295220

Reply:
Ah! TIL!